Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -92,26 +93,43 @@ public:
             WARPX_ABORT_WITH_MESSAGE("Binary collision " + collision_name + " must have exactly two species.");
         }
 
+        const CollisionType collision_type = BinaryCollisionUtils::get_collision_type(collision_name, mypc);
+
         m_isSameSpecies = (m_species_names[0] == m_species_names[1]);
 
-        m_binary_collision_functor = CollisionFunctorType(collision_name, mypc, m_isSameSpecies);
+        m_binary_collision_functor = CollisionFunctor(collision_name, mypc, m_isSameSpecies);
 
         const amrex::ParmParse pp_collision_name(collision_name);
         pp_collision_name.queryarr("product_species", m_product_species);
+
+        // if DSMC the colliding species are also product species
+        if (collision_type == CollisionType::DSMC) {
+            m_product_species.push_back(m_species_names[0]);
+            if (!m_isSameSpecies) {
+                m_product_species.push_back(m_species_names[1]);
+            }
+            // rotate to have the colliding species first
```

Comment:
I don't think I understand why this is needed.
Also, wouldn't it be simpler to do:
```
            
            if (m_isSameSpecies) {
                m_product_species.push_back(m_species_names[0]);
            } else {
                // have the colliding species first
                m_product_species.push_back(m_species_names[1]);
                m_product_species.push_back(m_species_names[0]);
            }
```

Reply:
without a call to `rotate`

Reply:
Basically, since the DSMC scheme splits particles during collision (to account for differences in weight) the colliding species are also considered product species. That is why the colliding species are added to the `m_product_species` vector. The `SplitAndScatterFunc` operator is written such that it assumes `m_species_names[0]` is also the 0'th element of `m_product_species` (and similarly for `m_species_names[1]` if more than 1 species is involved). The `rotate` is needed since there could be other product species in future DSMC processes (ionization for example).
Of course the whole thing could be changed so that `SplitAndScatterFunc` uses the last (or last two) element(s), but I don't think it would make a difference.

Reply:
Ah, I get it now, thanks.